server/src/main/java/io/crate/types/NumericStorage.java

```diff
@@ -131,29 +132,33 @@ public void indexValue(@NotNull BigDecimal value, IndexDocumentBuilder docBuilde
         }
     }
 
-    private static class FixedNumericIndexer extends BaseNumericIndexer {
+    private static class LargeNumericIndexer extends BaseNumericIndexer {
 
         private final FieldType fieldType;
+        private final int maxBytes;
 
-        private FixedNumericIndexer(Reference ref, int precision) {
+        private LargeNumericIndexer(Reference ref, NumericType type, int precision) {
             super(ref);
+            this.maxBytes = type.maxBytes();
             this.fieldType = new FieldType();
```

Comment:
`BigIntegerPoint` only sets:

```
  private static FieldType getType(int numDims) {
    FieldType type = new FieldType();
    type.setDimensions(numDims, BYTES);
    type.freeze();
    return type;
  }
```

Do you think it makes sense to omit the other setters too?  Some of the defaults are different (omitNorms=false, IndexOptions.NONE, tokenized=true)

Reply:
We should omit them, yes.  In particular, we need to make sure index options is `IndexOptions.NONE` - `IndexOptions.DOCS` means that we'll build a terms index as well as a points index, which we don't want here.  Norms and tokenization only apply to field types with a terms index so it doesn't matter what we set them to.